x-pack/plugins/security/server/lib/authentication/__tests__/authenticator.js

```diff
@@ -357,6 +356,35 @@ describe('Authenticator', () => {
       sinon.assert.calledWithExactly(session.clear, notSystemAPIRequest);
     });
 
+    it('clears session if provider requested it via setting state to `null`.', async () => {
```

Comment:
__note:__ these tests are more like integration ones and I don't really like that, but not didn't want to change them too much in this PR.